src/Microsoft.DotNet.Interactive.Mermaid/MermaidExtensions.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text;
+using System.Text.Encodings.Web;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Html;
+using Microsoft.DotNet.Interactive.Formatting;
+using Microsoft.DotNet.Interactive.Http;
+
+namespace Microsoft.DotNet.Interactive.Mermaid;
+
+public static class MermaidExtensions
+{
+
+    public static T UseMermaid<T>(this T kernel, Uri? libraryUri = null, string? libraryVersion = null, string? cacheBuster = null) where T : Kernel
+    {
+        cacheBuster = string.IsNullOrWhiteSpace(cacheBuster) ? Guid.NewGuid().ToString("N") : cacheBuster;
+        Formatter.Register<MermaidMarkdown>((markdown, writer) =>
```

Comment:
Since we own the type, I'd recommend using the attribute-based registration.

Reply:
how would that be reconfigurable?

Reply:
using the ITypedFormatter approach but left the `UseMermaid` as way to configure the library url and library version